//go:build !disable_pgv
// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: envoy/extensions/grpc_service/call_credentials/access_token/v3/access_token_credentials.proto

package access_tokenv3

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on AccessTokenCredentials with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *AccessTokenCredentials) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on AccessTokenCredentials with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// AccessTokenCredentialsMultiError, or nil if none found.
func (m *AccessTokenCredentials) ValidateAll() error {
	return m.validate(true)
}

func (m *AccessTokenCredentials) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Token

	if len(errors) > 0 {
		return AccessTokenCredentialsMultiError(errors)
	}

	return nil
}

// AccessTokenCredentialsMultiError is an error wrapping multiple validation
// errors returned by AccessTokenCredentials.ValidateAll() if the designated
// constraints aren't met.
type AccessTokenCredentialsMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m AccessTokenCredentialsMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m AccessTokenCredentialsMultiError) AllErrors() []error { return m }

// AccessTokenCredentialsValidationError is the validation error returned by
// AccessTokenCredentials.Validate if the designated constraints aren't met.
type AccessTokenCredentialsValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e AccessTokenCredentialsValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e AccessTokenCredentialsValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e AccessTokenCredentialsValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e AccessTokenCredentialsValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e AccessTokenCredentialsValidationError) ErrorName() string {
	return "AccessTokenCredentialsValidationError"
}

// Error satisfies the builtin error interface
func (e AccessTokenCredentialsValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sAccessTokenCredentials.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = AccessTokenCredentialsValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = AccessTokenCredentialsValidationError{}
